x-pack/plugins/ml/public/data_frame/pages/job_management/components/job_list/job_service/delete_job.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { toastNotifications } from 'ui/notify';
+import { ml } from '../../../../../../services/ml_api_service';
+
+import { DataFrameJobListRow } from '../common';
+
+export const deleteJobFactory = (getJobs: () => void) => async (d: DataFrameJobListRow) => {
```

Comment:
I'd say `() => () => {}` is not an uncommon pattern to do function factories in ES6 but I agree it might be tricky to read with all the intermingled TS. I tend to leave it in as is for this PR though but happy to refactor in one of the follow ups.